pallets/proofs/src/crypto/groth16.rs

```diff
@@ -161,31 +157,9 @@ where
     let mut rand_z: Vec<_> = Vec::with_capacity(proof_num);
     let mut accum_y = E::Fr::ZERO;
 
-    // TODO(@th7nder,#810, 18/03/2025): this is unsafe! randomness must be fetched from the chain.
-    use rand::Rng;
-    use rand_xorshift::XorShiftRng;
-    let rng = &mut XorShiftRng::from_seed([
-        0x59, 0x62, 0xbe, 0x5d, 0x76, 0x3d, 0xd, 0x8d, 0x17, 0xdb, 0x37, 0x32, 0x54, 0x06, 0xbc,
-        0xe5,
-    ]);
-
+    let mut rng = seeded_rng(seed);
     for _ in 0..proof_num {
-        let t: u128 = rng.gen();
-
-        let mut repr = E::Fr::ZERO.to_repr();
-        let mut repr_u64s = le_bytes_to_u64s(repr.as_ref());
-        assert!(repr_u64s.len() > 1);
-
-        repr_u64s[0] = (t & (-1i64 as u128) >> 64) as u64;
-        repr_u64s[1] = (t >> 64) as u64;
-
-        for (i, limb) in repr_u64s.iter().enumerate() {
-            let start = i * 8;
-            let stop = start + 8;
-            repr.as_mut()[start..stop].copy_from_slice(&limb.to_le_bytes());
-        }
-
-        let fr = E::Fr::from_repr(repr).unwrap();
+        let fr = E::Fr::random(&mut rng);
```

Comment:
What gives with this major change?

Reply:
This is effectively doing the same thing but in a simpler way.